Products/ZenEvents/tests/test_zentrap.py

```diff
@@ -70,8 +76,670 @@ def test_decode_unexpected_object_type(self):
+        oidMap = {
+            "1.2": "Zenoss",
+            "1.2.3": "Zenoss.Test",
+            "1.2.3.2": "Zenoss.Test.inexactMatch"
+        }
+        task = MockTrapTask(oidMap)
+        result = task.oid2name(".1.2.3.2.5", exactMatch=False, strip=True)
+        self.assertEqual(result, "Zenoss.Test.inexactMatch")
+        result = task.oid2name(".1.2.3.2.5.6", exactMatch=False, strip=True)
+        self.assertEqual(result, "Zenoss.Test.inexactMatch")
+
+    def test_AcceptsTuple(self):
+        oidMap = {}
+        task = MockTrapTask(oidMap)
+        self.assertEqual(task.oid2name((1, 2, 3, 4)), "1.2.3.4")
+
+
+class TestDecodeSnmpV1(BaseTestCase):
+
+    def makePacket(self):
+        pckt = FakePacket()
+        pckt.version = SNMPv1
+        pckt.host = "localhost"
+        pckt.port = 162
+        pckt.variables = ()
+        pckt.community = ""
+        pckt.enterprise_length = 0
+
+        # extra fields for SNMPv1 packets
+        pckt.agent_addr = [192, 168, 24, 4]
+        pckt.trap_type = 6
+        pckt.specific_type = 5
+        pckt.enterprise = "1.2.3.4"
+        pckt.enterprise_length = len(pckt.enterprise)
+        pckt.community = "community"
+        return pckt
+
+    def test_SnmpVersion(self):
+        pckt = self.makePacket()
+        task = MockTrapTask({})
+        eventType, result = task.decodeSnmpv1(("localhost", 162), pckt)
+        self.assertEqual(result["snmpVersion"], "1")
+
+    def test_NoAgentAddr(self):
+        pckt = self.makePacket()
+        del pckt.agent_addr
+        task = MockTrapTask({})
+        eventType, result = task.decodeSnmpv1(("localhost", 162), pckt)
+        self.assertEqual(result["device"], "localhost")
+
+    def test_HasAgentAddr(self):
+        pckt = self.makePacket()
+        task = MockTrapTask({})
+        eventType, result = task.decodeSnmpv1(("localhost", 162), pckt)
+        self.assertEqual(result["device"], "192.168.24.4")
+
+    def test_SnmpV1Enterprise(self):
+        pckt = self.makePacket()
+        task = MockTrapTask({})
+        eventType, result = task.decodeSnmpv1(("localhost", 162), pckt)
+        self.assertEqual(result["snmpV1Enterprise"], "1.2.3.4")
+
+    def test_SnmpV1GenericTrapType(self):
+        pckt = self.makePacket()
+        task = MockTrapTask({})
+        eventType, result = task.decodeSnmpv1(("localhost", 162), pckt)
+        self.assertEqual(result["snmpV1GenericTrapType"], 6)
+
+    def test_SnmpV1SpecificTrap(self):
+        pckt = self.makePacket()
+        task = MockTrapTask({})
+        eventType, result = task.decodeSnmpv1(("localhost", 162), pckt)
+        self.assertEqual(result["snmpV1SpecificTrap"], 5)
+
+    def test_EnterpriseOIDWithoutExtraZero(self):
+        pckt = self.makePacket()
+        task = MockTrapTask({})
+        eventType, result = task.decodeSnmpv1(("localhost", 162), pckt)
+        self.assertEqual(eventType, "1.2.3.4.5")
+        self.assertEqual(result["oid"], "1.2.3.4.5")
```

Comment:
I would combine these tests (line 158 - 200), because they are all testing the same input to the function under test, and validating attributes on the same result that it returns.

You could create an expected_result dictionary, and assert that it is equal to the returned result, but this would make the test more brittle.  It can be a good way to ensure future developers review these tests when they make a change to the method's result value.